ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/rest/RestBuilderClientTest.java

```diff
@@ -225,6 +236,31 @@ void registerValidators_failures() {
     verifyPostRequest("/eth/v1/builder/validators", signedValidatorRegistrationsRequest);
   }
 
+  @TestTemplate
```

Comment:
Are we missing the test when `setUserAgentHeader = true`?

Reply:
By default it is true, so have added USER_AGENT_HEADER_ASSERTIONS to the other tests.